engine/src/main/java/org/pentaho/di/trans/steps/loadfileinput/LoadFileInput.java

```diff
@@ -0,0 +1,525 @@
+        }
+        if ( meta.getUriField() != null && meta.getUriField().length() > 0 ) {
+          data.uriName = Const.optionallyDecodeUriString( data.file.getName().getURI() );
+        }
+        if ( meta.getRootUriField() != null && meta.getRootUriField().length() > 0 ) {
+          data.rootUriName = data.file.getName().getRootURI();
+        }
+        // get File content
+        getFileContent();
+
+        addFileToResultFilesName( data.file );
+
+        if ( isDetailed() ) {
+          logDetailed( BaseMessages.getString( PKG, "LoadFileInput.Log.FileOpened", data.file.toString() ) );
+        }
+      }
+
+    } catch ( Exception e ) {
+      logError( BaseMessages.getString( PKG, "LoadFileInput.Log.UnableToOpenFile", "" + data.filenr, data.file
+        .toString(), e.toString() ) );
+      stopAll();
+      setErrors( 1 );
+      return false;
+    }
+    return true;
+  }
+
+  public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws KettleException {
+    try {
+      // Grab a row
+      Object[] outputRowData = getOneRow();
+      if ( outputRowData == null ) {
+        setOutputDone(); // signal end to receiver(s)
+        return false; // end of data or error.
+      }
+
+      if ( isRowLevel() ) {
+        logRowlevel( BaseMessages.getString( PKG, "LoadFileInput.Log.ReadRow", data.outputRowMeta
+          .getString( outputRowData ) ) );
+      }
+
+      putRow( data.outputRowMeta, outputRowData );
+
+      if ( meta.getRowLimit() > 0 && data.rownr > meta.getRowLimit() ) { // limit has been reached: stop now.
+        setOutputDone();
+        return false;
+      }
+    } catch ( KettleException e ) {
+      logError( BaseMessages.getString( PKG, "LoadFileInput.ErrorInStepRunning", e.getMessage() ) );
+      logError( Const.getStackTracker( e ) );
+      setErrors( 1 );
+      stopAll();
+      setOutputDone(); // signal end to receiver(s)
+      return false;
+    }
+    return true;
+
+  }
+
+  void getFileContent() throws KettleException {
+    try {
+      data.filecontent = getFileBinaryContent( data.file.toString() );
+    } catch ( java.lang.OutOfMemoryError o ) {
+      logError( "There is no enaugh memory to load the content of the file [" + data.file.getName() + "]" );
+      throw new KettleException( o );
+    } catch ( Exception e ) {
+      throw new KettleException( e );
+    }
+  }
+
+  /**
+   * Read a file.
+   *
+   * @param vfsFilename
+   *          the filename or URL to read from
+   * @return The content of the file as a byte[]
+   * @throws KettleException
+   */
+  public static byte[] getFileBinaryContent( String vfsFilename ) throws KettleException {
+    InputStream inputStream = null;
+
+    byte[] retval = null;
+    try {
+      inputStream = KettleVFS.getInputStream( vfsFilename );
+      retval = IOUtils.toByteArray( new BufferedInputStream( inputStream ) );
+    } catch ( Exception e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 282.